plugins/com.google.cloud.tools.eclipse.aeri/src/com/google/cloud/tools/eclipse/aeri/ExceptionSender.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.aeri;
+
+import com.google.cloud.tools.eclipse.util.CloudToolsInfo;
+import com.google.cloud.tools.eclipse.util.io.HttpUtil;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.eclipse.epp.logging.aeri.core.IStackTraceElement;
+import org.eclipse.epp.logging.aeri.core.IThrowable;
+
+public class ExceptionSender {
+
+  private static final Logger logger = Logger.getLogger(ExceptionSender.class.getName());
+
+  private static final String REPORT_URL = "https://clients2.google.com/cr/report";
+
+  private static final String NONE_MARKER = "__NONE__";
+
+  private final String endpointUrl;
+
+  ExceptionSender() {
+    this(REPORT_URL);
+  }
+
+  @VisibleForTesting
+  ExceptionSender(String endpointUrl) {
+    this.endpointUrl = endpointUrl;
+  }
+
+  void sendException(IThrowable exception, String eclipseBuildId, String javaVersion,
+      String os, String osVersion, String userSeverity, String userComment) {
+    Map<String, String> parameters = new HashMap<>();
+    parameters.put("product", CloudToolsInfo.EXCEPTION_REPORT_PRODUCT_ID);
+    parameters.put("version", CloudToolsInfo.getToolsVersion());
+    parameters.put("exception_info", formatStacktrace(exception));
+
+    String extraInfo = "eclipseBuildId: " + nullOrEmptyToNone(eclipseBuildId) + "\n"
+                     + "javaVersion: " + nullOrEmptyToNone(javaVersion) + "\n"
+                     + "os: " + os + " " + nullOrEmptyToNone(osVersion) + "\n"
+                     + "userSeverity: " + nullOrEmptyToNone(userSeverity) + "\n"
+                     + "userComment: " + nullOrEmptyToNone(userComment);
+    // "comments" seems to be the only viable place where we can put extra info.
+    parameters.put("comments", extraInfo);
+
+    try {
+      // Internal system expects the product ID and version in URL too.
+      String urlParameters = "?product=" + CloudToolsInfo.EXCEPTION_REPORT_PRODUCT_ID
+          + "&version=" + CloudToolsInfo.getToolsVersion();
+      int code = HttpUtil.sendPostMultipart(endpointUrl + urlParameters, parameters);
+      if (code != 200) {
```

Comment:
Good idea, but no need to pull google-http-java-client into this bundle. I will use `java.net.HttpURLConnection.HTTP_OK`.